.github/workflows/quick.yaml

```
@@ -16,6 +16,12 @@ on:
     # test PRs targeting this branch code
     branches: [ "master" ]
 
+concurrency:
+  # Cancel ongoing tests in case of push to the same PR or staging branch,
+  # but let previous master commit tests complete.
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
```

I do not insist on this change, but current PR logic seems to apply to all stale jobs (i.e. jobs testing a now-stale PR or staged commit), _regardless_ of which workflow they were launched from.

```suggestion
  group: ${{ github.event.pull_request.number || github.ref }}
```

If this is accepted, PR title should be adjusted to remove the word "workflow".

---

Accepted practice consistently keys on workflow; it feel dangerous.
What if Anubis decided to abort a staging test while a followup PR test is being executed?
Would the followup PR test not be cancelled, resulting in a stall?

---

> Accepted practice consistently keys on workflow

I do not see enough evidence to support that claim, but I agree that portions of GitHub [documentation](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#concurrency) can be interpreted as disallowing running (or even queuing) tests for _two workflows_ with the same concurrency key when `cancel-in-progress` is set to true. Our slow/fast split/optimization may be working against GitHub concurrency support design here. If this keeps happening, we may have to redesign that optimization to move _all_ staging tests to one workflow (but it is probably impossible to do that without preprocessing or duplicating YAML).

AFAICT, keeping github.workflow in the concurrency group key would still cancel stale staging tests (they would just be running in two different groups, but both groups would be cancelled), and I realize that it is difficult to test these things, so I am resolving my change request.